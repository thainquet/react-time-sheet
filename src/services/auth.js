import axios from 'axios'

const register = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/register`, params);
}

const login = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/login`, params);
}

const changePassword = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/changePass`, params);
}

const forgotPassword = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/forgotPass`, params);
}

export default {
  login, changePassword, forgotPassword, register
}