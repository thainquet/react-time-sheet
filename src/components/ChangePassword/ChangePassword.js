import React, { useState } from 'reactn';
import { isLogin } from 'helpers/auth'
import { authServices } from 'services'


const ChangePass = props => {
  const { initFormParams = {
    username: '',
    password: '',
    newpassword: ''
  } } = props;
  if (!isLogin()) props.history.push('/login')
  const [form, setFormInfo] = useState(initFormParams);

  const [res, resMessage] = useState('')

  const resetPass = async () => {
    resMessage(res => res = '')
    if (!form.username || !form.password || !form.newpassword)
      resMessage(res => res = 'All fields required!')
    else
      if (form.password === form.newpassword)
        resMessage(res => res = 'Old password and new password must be different!')
      else {
        const params = {
          username: form.username,
          password: form.password,
          newpassword: form.newpassword
        }
        const result = await authServices.changePassword(params)
        if (result) resMessage(res => res = result.data.message)
      }
  }

  const updateField = event => {
    setFormInfo({
      ...form,
      [event.target.name]: event.target.value
    });
  };

  return (
    <div>
      <div>
        <label htmlFor="">Username</label>
        <input type="text" name="username" id="username" autoFocus autoComplete="off" value={form.username} onChange={updateField} />
      </div>

      <div>
        <label htmlFor="">Password</label>
        <input type="password" name="password" id="password" autoComplete="off" value={form.password} onChange={updateField} />
        <br />
      </div>

      <div>
        <label htmlFor="">New password</label>
        <input type="password" name="newpassword" id="newpassword" autoComplete="off" value={form.newpassword} onChange={updateField} />
        <br />
      </div>

      <pre>{res}</pre>

      <div>
        <button onClick={resetPass} id="change-passw-btn">Change Password</button>
        <button onClick={() => props.history.push('/home')}>Back to Homepage</button>
      </div>
    </div>
  )
}

export default ChangePass;

