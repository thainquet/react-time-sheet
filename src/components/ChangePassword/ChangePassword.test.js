import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { authServices } from 'services';
import ChangePassword from './ChangePassword';

const props = {
  initFormParams: {
    username: 'thainq00',
    password: '1',
    newpassword: '12'
  },
  history: [],
  res: ''
}

it('should render correctly', () => {
  const wrapper = renderer.create(<ChangePassword {...props} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('has 3 inputs and 2 buttons', () => {
  const wrapper = mount(<ChangePassword {...props} />);
  expect(wrapper.find('input')).toHaveLength(3);
  expect(wrapper.find('button')).toHaveLength(2);
  wrapper.unmount();
});

it('change password success', () => {
  const wrapper = mount(<ChangePassword {...props} />);
  const changePasswordFunc = jest.spyOn(authServices, 'changePassword').mockImplementation(() => Promise.resolve({ data: { code: 200, message: 'Thanh cong' } }));
  // wrapper.find('input#username').simulate('change', { target: { value: props.initFormParams.username } })
  // wrapper.find('input#password').simulate('change', { target: { value: props.initFormParams.password } })
  // wrapper.find('input#newpassword').simulate('change', { target: { value: props.initFormParams.newpassword } })
  wrapper.find('button#change-passw-btn').simulate('click')
  expect(changePasswordFunc).toHaveBeenCalledWith(
    props.initFormParams
  );
  wrapper.unmount();
});
