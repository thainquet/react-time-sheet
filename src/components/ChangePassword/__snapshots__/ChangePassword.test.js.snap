// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div>
    <label
      htmlFor=""
    >
      Username
    </label>
    <input
      autoComplete="off"
      autoFocus={true}
      id="username"
      name="username"
      onChange={[Function]}
      type="text"
      value="thainq00"
    />
  </div>
  <div>
    <label
      htmlFor=""
    >
      Password
    </label>
    <input
      autoComplete="off"
      id="password"
      name="password"
      onChange={[Function]}
      type="password"
      value="1"
    />
    <br />
  </div>
  <div>
    <label
      htmlFor=""
    >
      New password
    </label>
    <input
      autoComplete="off"
      id="newpassword"
      name="newpassword"
      onChange={[Function]}
      type="password"
      value="12"
    />
    <br />
  </div>
  <pre>
    
  </pre>
  <div>
    <button
      id="change-passw-btn"
      onClick={[Function]}
    >
      Change Password
    </button>
    <button
      onClick={[Function]}
    >
      Back to Homepage
    </button>
  </div>
</div>
`;
